demo/screens/ListLands.js

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import {
+  ListView,
+  Image,
+  StyleSheet,
+  View,
+  Text,
+  Button,
+  TouchableHighlight,
+} from "react-native";
+import Land from "../components/presentational/Land";
+
+const rows = [
+  {
+    id: 1,
+    name: "DRAGON KINGDOM",
+    img: require("../assets/images/lands/land1.png"),
+    priceMana: 13999,
+    priceUsd: 699,
+  },
+  {
+    id: 2,
+    name: "Parcel",
+    img: require("../assets/images/lands/land2.png"),
+    priceMana: 27985,
+    priceUsd: 1399,
+  },
+  {
+    id: 3,
+    name: "Golden Area",
+    img: require("../assets/images/lands/land3.png"),
+    priceMana: 19000,
+    priceUsd: 950,
+  },
+  {
+    id: 4,
+    name: "2 parcels away from Main Road to Genesis Plaza",
+    img: require("../assets/images/lands/land4.png"),
+    priceMana: 150000,
+    priceUsd: 7500,
+  },
+];
+
+const rowHasChanged = (r1, r2) => r1.id !== r2.id;
+
+const ds = new ListView.DataSource({ rowHasChanged });
+
+export default class ListLands extends React.Component {
+  state = {
+    dataSource: ds.cloneWithRows(rows),
+  };
+
+  renderRow = rowData => {
+    const { id, name, img, priceMana, priceUsd } = rowData;
+    return (
+      <TouchableHighlight
```

Comment:
I think this could happen inside a component that ListLands uses in its renderRow function

Reply:
It's not clear to me, because It's already inside of renderRow function.

Reply:
I was thinking one component defined above ListLands that is then used in renderRow. I'd have to actually code it to make sure all the data is passed correctly and available where it needs to be, but something like this:
```
function LandRow(props) {
  ...
}
...
export default class ListLands extends React.Component { 
...
  renderRow = rowData => {
    const handlePress={() =>
      this.props.navigation.navigate("LandClaim", { land: rowData })
    }
    return (
      <LandRow rowData=rowData onPress=handlePress>
    );
  );
```